VimeoDeeplink/VimeoDeeplink/VIMDeeplink.h

```diff
@@ -0,0 +1,62 @@
+//
+//  VIMDeeplink.h
+//  Vimeo
+//
+//  Created by Alfred Hanssen on 9/10/14.
+//  Copyright (c) 2014 Vimeo. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+@import Foundation;
+
+extern NSString * _Nonnull VimeoBaseURLString;
+
+extern NSString * _Nonnull VimeoAppStoreURLString;
+
+@interface VIMDeeplink: NSObject
+
++ (BOOL)viewVimeoAppInAppStore;
++ (BOOL)isVimeoAppInstalled;
++ (BOOL)openVimeo;
++ (BOOL)openCategories;
++ (BOOL)openCategory:(NSString * _Nonnull)uri;
++ (BOOL)openChannel:(NSString * _Nonnull)uri;
++ (BOOL)openFeed;
++ (BOOL)openExplore;
++ (BOOL)openLibrary;
++ (BOOL)openMe;
++ (BOOL)openUpload;
++ (BOOL)openOffline;
++ (BOOL)openPurchases;
++ (BOOL)openWatchLater;
++ (BOOL)openSearch;
++ (BOOL)openSearchVideos;
++ (BOOL)openSearchChannels;
++ (BOOL)openSearchPeople;
++ (BOOL)openVideoPlayer:(NSString * _Nonnull)uri;
++ (BOOL)openVideoPlayerUpNextTab:(NSString * _Nonnull)uri;
++ (BOOL)openVideoPlayerCommentsTab:(NSString * _Nonnull)uri;
++ (BOOL)openVideoPlayerLikesTab:(NSString * _Nonnull)uri;
++ (BOOL)openUserProfile:(NSString * _Nonnull)uri;
++ (BOOL)openUserFollowers:(NSString * _Nonnull)uri;
++ (BOOL)openUserFollowing:(NSString * _Nonnull)uri;
++ (BOOL)openUserLikes:(NSString * _Nonnull)uri;
```

Comment:
I guess I don't see how that would be helpful to people. HTTP/S URLs are opened automatically in the app via "App Index" and "Universal Link" functionality when a user taps on an HTTP/S URL anywhere in the OS. 

If we were to someone wrap this here it would look like this: 

```Swift
public static func openUrl(url: NSURL) -> Bool
{
    return UIApplication.sharedApplication.openURL(url)
}
```

The user might as well just call `UIApplication.sharedApplication.openURL(url)` on their own. 

If we somehow convert the URL into a deeplink with "vimeo" as the scheme we're duplicating OS functionality which would be dangerous. 

Maybe there's something I'm missing?